packages/salesforcedx-vscode-apex-replay-debugger/src/index.ts

```diff
@@ -44,24 +46,53 @@ function registerCommands(): vscode.Disposable {
         defaultUri: getDialogStartingPath()
       });
       if (fileUris && fileUris.length === 1) {
-        lastOpenedLogFolder = path.dirname(fileUris[0].fsPath);
+        extContext.workspaceState.update(
```

Comment:
If you extract this out to its own method, then (1) you can re-use it in `launchFromLogFileCmd` below and (2) you can add a little test for it. The method can take an ExtensionContext and a string path.

Your test can do something like https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/packages/salesforcedx-vscode-lwc-next/test/eslint.test.ts#L16. You're defining a context object and its update method which can simply add whatever key value it gets to an array or something. Then you assert the array contains LAST_OPENED_LOG_KEY, LAST_OPENED_LOG_FOLDER_KEY and their values. The test will verify the production method uses the context object to save those key value pairs.